receiver/awscontainerinsightreceiver/internal/stores/store.go

```diff
@@ -50,17 +50,19 @@ type K8sDecorator struct {
 	// It would be easier to keep the ctx here than passing it as a parameter for Decorate(...) function.
 	// The K8sStore (e.g. podstore) does network request in Decorate function, thus needs to take a context
 	// object for canceling the request
-	ctx context.Context
+	ctx                         context.Context
+	addContainerNameMetricLabel bool
 }
 
-func NewK8sDecorator(ctx context.Context, tagService bool, prefFullPodName bool, addFullPodNameMetricLabel bool, logger *zap.Logger) (*K8sDecorator, error) {
+func NewK8sDecorator(ctx context.Context, tagService bool, prefFullPodName bool, addFullPodNameMetricLabel bool, addContainerNameMetricLabel bool, logger *zap.Logger) (*K8sDecorator, error) {
```

Comment:
There are a lot of boolean flags being appended to this. Should we just compose them all inside of a config struct to maintain some stability on this function definition?

Reply:
not part of this commit